lib/rtcpeerconnection/safari.js

```diff
@@ -236,24 +236,15 @@ SafariRTCPeerConnection.prototype.close = function close() {
   });
 };
 
-SafariRTCPeerConnection.prototype.addStream = function addStream(stream) {
-  if (this._localStreams.has(stream)) {
-    return;
-  }
-  var tracks = new Set(stream.getTracks());
-  this._peerConnection.getSenders().forEach(function(sender) {
-    if (tracks.has(sender.track)) {
-      tracks.delete(sender.track);
-    }
-  });
-  tracks.forEach(function(track) {
-    try {
-      this.addTrack(track, stream);
-    } catch (error) {
-      // Do nothing.
-    }
-  }, this);
-  this._localStreams.set(stream, new Set(tracks));
+// NOTE(mmalavalli): webrtc-adapter has a bug where the "addTrack" shim
+// does not return an RTCRtpSender and returns undefined instead. An issue
+// [https://github.com/webrtc/adapter/issues/714] has been filed, and this
+// shim will be removed once it is addressed.
+SafariRTCPeerConnection.prototype.addTrack = function addTrack() {
+  var args = [].slice.call(arguments);
+  var mediaStreamTrack = args[0];
+  var sender = this._peerConnection.addTrack.apply(this._peerConnection, args);
+  return sender || { track: mediaStreamTrack };
 };
```

Comment:
Same comments/questions as above:

1. Can we leave `addStream` as-is?
2. Should we shim an RTCRtpSender implementation rather than return an ad-hoc object?

Reply:
@markandrus +1 for shimming `RTCRtpSender`. However, I don't get why we need to retain the legacy shims.

Reply:
I guess you're right 👍 we do not need to shim the legacy APIs.